modules/instrument_list/php/module.class.inc

```diff
@@ -63,26 +63,46 @@ class Module extends \Module
 
         $gets = $request->getQueryParams();
 
-        $attribute = $request->getAttribute("CandID");
-        if ($attribute !== null) {
-            $candidate = \Candidate::singleton($attribute);
-        } else {
-            $candidate = \Candidate::singleton($gets['candID'] ?? '');
-        }
-        $request = $request->withAttribute("Candidate", $candidate);
+        try {
+            $attribute = $request->getAttribute("CandID");
+            if ($attribute !== null) {
+                $candidate = \Candidate::singleton(intval($attribute));
+            } else {
+                $candidate = \Candidate::singleton(intval($gets['candID']) ?? -1);
+            }
+            $request = $request->withAttribute("Candidate", $candidate);
 
-        $attribute = $request->getAttribute("timePoint");
-        if ($attribute === null) {
-            $request = $request->withAttribute(
-                "timePoint",
-                \TimePoint::singleton($gets['sessionID'])
-            );
-        }
+            $attribute = $request->getAttribute("timePoint");
+            if ($attribute === null) {
+                $request = $request->withAttribute(
+                    "timePoint",
+                    \TimePoint::singleton(intval($gets['sessionID']))
+                );
+            }
 
-        // We need to set the internal page properties for hasAccess to succeed.
-        $page->setCandidate($candidate);
-        $page->setTimePoint($request->getAttribute("timePoint"));
+            // We need to set the internal page properties for hasAccess to succeed.
+            $page->setCandidate($candidate);
+            $page->setTimePoint($request->getAttribute("timePoint"));
 
+        } catch (\LorisException $e) {
+            $user    = $request->getAttribute("user") ?? new \LORIS\AnonymousUser();
+            $message = 'Could not find candidate information for candidate '
+                . htmlspecialchars($gets['candID']) . ' and session '
+                . htmlspecialchars($gets['sessionID']) . '.';
+            return (new \LORIS\Middleware\PageDecorationMiddleware(
+                $user
+            ))->process(
+                $request,
+                new \LORIS\Router\NoopResponder(
+                    new \LORIS\Http\Error(
+                        $request,
+                        500,
```

Comment:
If this exception is caused by not being able to find the candidate/session, that's a 404 (user trying to access a non-existent resource), not a 500 (server error). (But if there are other causes of the exception, then the error message is misleading..)